lib/shopping_cart.rb

```diff
@@ -0,0 +1,26 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+  def initialize(name,capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def total_number_of_products
+    (products[0].quantity.to_i)+(products[1].quantity.to_i)+(products[2].quantity.to_i)
+  end
```

Comment:
This isn't quite working, because it will only work for three products - what if the shopping cart has 4 or 10 or 100 products?  You should use either `each` or some other enumerable to iterate over and add up all the products' quantities (a handy method here is `sum`)